src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -164,41 +170,44 @@ private void VerifyConnected()
 				// need to try IP Addresses one at a time: https://github.com/dotnet/corefx/issues/5829
 				foreach (var ipAddress in ipAddresses)
 				{
-					cancellationToken.ThrowIfCancellationRequested();
-
-					Socket socket = null;
+					var socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
 					try
 					{
-						socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
-						using (cancellationToken.Register(() => socket.Dispose()))
+						var connectTasks = new List<Task>
```

Comment:
What was wrong with registering a handle on the cancellationToken that called `Socket.Dispose`?

As per [MSDN](https://msdn.microsoft.com/en-us/library/tad07yt6.aspx):

> To cancel a pending call to the BeginConnect method, close the Socket. When the Close method is called while an asynchronous operation is in progress, the callback provided to the BeginConnect method is called. A subsequent call to the EndConnect method will throw an ObjectDisposedException to indicate that the operation has been cancelled. 

(The Socket async API has lots of "oddities" and this is one of them.)